package-res/resources/web/pentaho/data/filter/Not.js

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./AbstractFilter",
+  "../../util/object",
+  "../../lang/ArgumentRequiredError",
+  "./_apply",
+  "./_toSpec"
+], function(AbstractFilter, O, ArgumentRequiredError, _apply, _toSpec) {
+  "use strict";
+
+  /**
+   * @name Not
+   * @memberOf pentaho.data.filter
+   * @class
+   * @throws {pentaho.lang.ArgumentRequiredError} When `operand` is not specified.
```

Comment:
I don’t understand…Do you mean like this:

@throws When `operand` is not specified.

From: Duarte Cunha Leão [mailto:notifications@github.com]
Sent: Tuesday, March 01, 2016 11:02 AM
To: pentaho/pentaho-platform-plugin-common-ui <pentaho-platform-plugin-common-ui@noreply.github.com>
Cc: Scott Aslan <saslan@pentaho.com>
Subject: Re: [pentaho-platform-plugin-common-ui] [BACKLOG-6419] Implementation and tests for Filters (for REVIEW purposes only) (#652)


In package-res/resources/web/pentaho/data/filter/Not.js<https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/652#discussion_r54588317>:

> + * limitations under the License.

> + */

> +define([

> +  "./AbstractFilter",

> +  "../../util/object",

> +  "../../lang/ArgumentRequiredError",

> +  "./_apply",

> +  "./_toSpec"

> +], function(AbstractFilter, O, ArgumentRequiredError, _apply, _toSpec) {

> +  "use strict";

> +

> +  /**

> +   * @name Not

> +   * @memberOf pentaho.data.filter

> +   * @class

> +   * @throws {pentaho.lang.ArgumentRequiredError} When `operand` is not specified.

Don't declare ArgumentRequiredError unless it is not always thrown (dynamic rule).

—
Reply to this email directly or view it on GitHub<https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/652/files#r54588317>.